main.tf

```diff
@@ -59,61 +52,46 @@ resource "aws_route_table" "public" {
   }
 }
 
-resource "aws_route_table_association" "public1" {
-  subnet_id      = aws_subnet.public1.id
+resource "aws_route_table_association" "public" {
+  count = local.count
+  subnet_id = aws_subnet.public[count.index].id
   route_table_id = aws_route_table.public.id
 }
 
-resource "aws_route_table_association" "public2" {
-  subnet_id      = aws_subnet.public2.id
-  route_table_id = aws_route_table.public.id
-}
-
-resource "aws_eip" "nat1" {
-  vpc      = true
-}
 
-resource "aws_eip" "nat2" {
+resource "aws_eip" "eip" {
+  count = local.count 
   vpc      = true
 }
 
-resource "aws_nat_gateway" "nat1" {
-  allocation_id = aws_eip.nat1.id
-  subnet_id     = aws_subnet.public1.id
+resource "aws_nat_gateway" "nat" {
+  count         =  2
```

Comment:
could you please change `count = 2` to `count =  local.count` so as to maintain consistent in your code